test/backend_graph_comparison.in.cpp

```diff
@@ -0,0 +1,278 @@
+//*****************************************************************************
+// Copyright 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <cstdlib>
+#include <string>
+#include "gtest/gtest.h"
+
+#include "ngraph/log.hpp"
+#include "ngraph/ngraph.hpp"
+#include "ngraph/runtime/host_tensor.hpp"
+#include "ngraph/serializer.hpp"
+#include "util/all_close.hpp"
+#include "util/all_close_f.hpp"
+#include "util/test_control.hpp"
+
+using namespace ngraph;
+
+static std::string s_manifest = "${MANIFEST}";
+
+// Params to drive concat_vector_large testing variations
+class frozen_graph_files : public ::testing::TestWithParam<std::string>
+{
+public:
+    void compare_results(NodeVector& result_nodes,
+                         std::vector<std::shared_ptr<runtime::Tensor>> ref_results,
+                         std::vector<std::shared_ptr<runtime::Tensor>> bk_results)
+    {
+        for (int i = 0; i < ref_results.size(); ++i)
+        {
+            const std::shared_ptr<runtime::Tensor>& ref_data = ref_results.at(i);
+            const std::shared_ptr<runtime::Tensor>& bk_data = bk_results.at(i);
+
+            std::cout << "Comparing results for " << result_nodes.at(i)->get_name() << std::endl;
+            if (auto node = std::dynamic_pointer_cast<op::GetOutputElement>(result_nodes.at(i)))
+            {
+                std::cout << "  Parent node: ";
+                for (auto& p : node->get_arguments())
+                {
+                    std::cout << " " << p->get_name() << std::endl;
+                    std::cout << "   nargs: " << p->get_arguments().size() << std::endl;
+                }
+            }
+
+            EXPECT_EQ(ref_data->get_element_type(), bk_data->get_element_type());
```

Comment:
Since I suspect very bad things happen if type/count/shapes do not match then you may want to ASSERT_EQ here

Reply:
Good point. Changed.